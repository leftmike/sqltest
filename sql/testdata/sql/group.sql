--
-- Test GROUP BY
--

DROP TABLE IF EXISTS tbl1;

CREATE TABLE tbl1 (c1 int, c2 int, c3 int, c4 int);

INSERT INTO tbl1 VALUES
    (1, 2, 3, 4),
    (1, 2, 33, 44),
    (2, 1, 3, NULL),
    (2, 2, 3, 4),
    (2, 3, 4, NULL),
    (3, 4, 5, 6),
    (4, 5, 6, 7),
    (4, 1, NULL, 5),
    (4, 1, 5, 6),
    (5, 1, 6, 7),
    (5, 2, NULL, NULL);

SELECT c1 FROM tbl1 GROUP BY c1;

SELECT c1 + c2 AS c FROM tbl1 GROUP BY c1 + c2;

SELECT c1 + c2 AS d1, (c1 + c2) * 10 AS d2 FROM tbl1 GROUP BY c1 + c2;

SELECT count(*) AS count_all, c1 + c2 AS c, count(c3), count(c4) FROM tbl1 GROUP BY c1 + c2;

SELECT c1, c2, count(*) AS count_all FROM tbl1 GROUP BY c1, c2;

SELECT c1, c2, count(*) AS count_all FROM tbl1 GROUP BY c2, c1;
