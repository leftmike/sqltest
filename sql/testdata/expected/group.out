--
-- Test GROUP BY
--
DROP TABLE IF EXISTS tbl1;
CREATE TABLE tbl1 (c1 int, c2 int, c3 int, c4 int);
INSERT INTO tbl1 VALUES
    (1, 2, 3, 4),
    (1, 2, 33, 44),
    (2, 1, 3, NULL),
    (2, 2, 3, 4),
    (2, 3, 4, NULL),
    (3, 4, 5, 6),
    (4, 5, 6, 7),
    (4, 1, NULL, 5),
    (4, 1, 5, 6),
    (5, 1, 6, 7),
    (5, 2, NULL, NULL);
SELECT c1 FROM tbl1 GROUP BY c1;
   c1
   --
 1  1
 2  2
 3  3
 4  4
 5  5
(5 rows)
SELECT c1 + c2 AS c FROM tbl1 GROUP BY c1 + c2;
   c
   -
 1 3
 2 4
 3 5
 4 6
 5 7
 6 9
(6 rows)
SELECT c1 + c2 AS d1, (c1 + c2) * 10 AS d2 FROM tbl1 GROUP BY c1 + c2;
   d1 d2
   -- --
 1  3 30
 2  4 40
 3  5 50
 4  6 60
 5  7 70
 6  9 90
(6 rows)
SELECT count(*) AS count_all, c1 + c2 AS c, count(c3), count(c4) FROM tbl1 GROUP BY c1 + c2;
   count_all c count count
   --------- - ----- -----
 1         1 4     1     1
 2         1 6     1     1
 3         1 9     1     1
 4         2 7     1     1
 5         3 3     3     2
 6         3 5     2     2
(6 rows)
SELECT c1, c2, count(*) AS count_all FROM tbl1 GROUP BY c1, c2;
   c1 c2 count_all
   -- -- ---------
 1  1  2         2
 2  2  1         1
 3  2  2         1
 4  2  3         1
 5  3  4         1
 6  4  1         2
 7  4  5         1
 8  5  1         1
 9  5  2         1
(9 rows)
SELECT c1, c2, count(*) AS count_all FROM tbl1 GROUP BY c2, c1;
   c1 c2 count_all
   -- -- ---------
 1  1  2         2
 2  2  1         1
 3  2  2         1
 4  2  3         1
 5  3  4         1
 6  4  1         2
 7  4  5         1
 8  5  1         1
 9  5  2         1
(9 rows)
